csharp/src/Ice/ColocatedConnector.cs

```diff
@@ -50,6 +52,8 @@ public Connection Connect(string connectionId)
 
         public override string ToString() =>
             _endpoint.Adapter.Name.Length == 0 ? "unnamed adapter" : _endpoint.Adapter.Name;
+        public override bool Equals(Connector? obj) => ReferenceEquals(Endpoint, obj?.Endpoint);
+        public override int GetHashCode() => _endpoint.GetHashCode();
```

Comment:
It can't hurt but it's not really useful given that the endpoint creates and caches the connector.

Reply:
isn't the hash still used when we add the connector to the factory lookup table?

Reply:
Right and the default hash/equal were ok because a given endpoint only creates a single connector. Anyway, let's compare the endpoints, it could be useful if one day we create multiple endpoint objects for a same OA.

Reply:
I implement it here because providing `==` and `!=` in the base Connector class warns about providing a custom Equals/GetHashCode implementation that I make abstract